exercises/numbers/power/power.js

```diff
@@ -21,7 +21,20 @@ function power(base, exponent) {
 
     That means you don't have to deal with, e.g., power(2, 1.5).
   */
- return base ** exponent
+  if (exponent === 0) {
```

Comment:
If you return early, you don't need the `else`. It'll make the code a little easier to follow.

```javascript
function power(base, exp) {
  if (exp === 0) {
    return 1;
  }

  if (base === 1) {
    return 1;
  }

  let result = 1;

  for (let i = 0; i < exponent; i++) {
    result *= base;
  }

  return result;
}
```

Reply:
But wait! If we handle it that way, we don't even need the `exponent === 0` check because we'll never enter the `for` loop:

```javascript
function power(base, exponent) {
  if (base === 1) {
    return 1;
  }

  let result = 1;

  for (let i = 0; i < exponent; i++) {
    result *= base;
  }

  return result;
}
```

Reply:
And even then, returning early if `base === 1` is just an optimization so that we don't pointlessly multiply `1` by itself a bunch of times. In other words, the following would still work:

```javascript
function power(base, exponent) {
  let result = 1;

  for (let i = 0; i < exponent; i++) {
    result *= base;
  }

  return result;
}
```

Reply:
If you want to play around, look into [exopnentiation by squaring](https://en.wikipedia.org/wiki/Exponentiation_by_squaring). This is a much faster way of squaring.

You can use the technique in https://github.com/jfarmer/adjacent-sessions/blob/master/2020-01-20/benchmark.js to compare the running times.